src/test/java/com/facebook/openwifirrm/modules/ModelerUtilsTest.java

```diff
@@ -77,4 +86,61 @@ void testValidMetric() throws Exception {
 		);
 		assertEquals(0.861, metric, 0.001);
 	}
+
+	@Test
+	void testAggregatedWifiScanEntry() {
+		final long obsoletionPeriodMs = 900000;
+
+		final String apA = "aaaaaaaaaaaa";
+		final String bssidA = "aa:aa:aa:aa:aa:aa";
+		final String apB = "bbbbbbbbbbbb";
+		final String bssidB = "bb:bb:bb:bb:bb:bb";
+		DeviceDataManager deviceDataManager = new DeviceDataManager();
+		deviceDataManager.setTopology(TestUtils.createTopology(TestUtils.TEST_ZONE, apA, apB));
+		DataModel dataModel = new DataModel();
+
+		// if there are no scans (no list or empty list), there should be no aggregates
+		dataModel.latestWifiScans.computeIfAbsent(bssidB, k -> new LinkedList<>());
+		assertTrue(ModelerUtils.getAggregatedWifiScans(dataModel, obsoletionPeriodMs, new MeanAggregator()).isEmpty());
+
+		// for an AP, if there is one scan, the aggregate should just be that scan
+		// for the other device, there should be no aggregate
+		int signal = -60;
+		String htOper = TestUtils.getHtOper();
+		String vhtOper = null;
+		long timeOffsetMs = 0;
+		ProcessedWifiScanEntry entryB1 = TestUtils.createWifiScanEntry(signal, bssidB, htOper, vhtOper, timeOffsetMs);
+		dataModel.latestWifiScans.get(bssidB).add(Arrays.asList(entryB1));
+		ProcessedWifiScanEntry aggregatedEntryB = ModelerUtils
+				.getAggregatedWifiScans(dataModel, obsoletionPeriodMs, new MeanAggregator()).get(bssidB);
+		assertFalse(dataModel.latestWifiScans.containsKey(bssidA));
+		assertEquals(entryB1, aggregatedEntryB);
+
+		// add another entry for device B and check the aggregation
+		signal = -62;
+		timeOffsetMs = 60000; // 1 min later
+		ProcessedWifiScanEntry entryB2 = TestUtils.createWifiScanEntry(signal, bssidB, htOper, vhtOper, timeOffsetMs);
+		dataModel.latestWifiScans.get(bssidB).add(Arrays.asList(entryB2));
+		aggregatedEntryB = ModelerUtils.getAggregatedWifiScans(dataModel, obsoletionPeriodMs, new MeanAggregator())
+				.get(bssidB);
+		ProcessedWifiScanEntry expectedAggregatedEntry = new ProcessedWifiScanEntry(entryB2);
+		expectedAggregatedEntry.signal = -61; // average of -60 and -62
+		assertFalse(dataModel.latestWifiScans.containsKey(bssidA));
+		assertEquals(expectedAggregatedEntry, aggregatedEntryB);
+
+		// test the obsoletion period
+		signal = -64;
+		timeOffsetMs = obsoletionPeriodMs; // now only entry 2 and entry 3 should be aggregated
```

Comment:
it's not clear to me why entry 2 (60 s) is in the obsoletion period and entry 1 (0 s) is not, if the period is 900 s.

Reply:
The obsoletion period is "exclusive" in that if the obsoletion period is X ms, then a record X ms ago is considered obsolete. That was just an arbitrary choice made by me (though I think it does avoid the need for a +/- 1 somewhere in the code so it is easier). I will make a clarifying comment.

Reply:
This will be handled in the new PR.